### ISSUE TITLE
Missing Documentation 1.2 / Motion Controller Mode

### ISSUE THREAD

**Issue by AlpineWhite** [bigtreetech/BIGTREETECH-TMC5160-V1.0#8]
Checking the documentation, there is none for version 1.2.  The motion controller mode also appears selectable via solder pads on version 1.0, but there is no reference as to where those pads are on the chip.

Can you please update the documentation for those of us who wish to use the motion controller mode?

**Comment by PMTaylor** [comment]
Agreed! I'm looking for this information.
Any leads @AlpineWhite ?

**Comment by PKav89** [comment]
Motion controller mode is not possible on v1.2. Pins 20, 21 and 22 (VCC_IO, SD_MODE and SPI_MODE) are tied together on the PCB. It is possible to cut this VERY SMALL trace and make a jumper wire from SD_MODE to GND, but it's rather difficult.

**Comment by PMTaylor** [comment]
Yeah, i'd figured this out by myself. Schematics on the site would be very helpful though, irrespective of the answer to this question.
If you do a V1.3, please return a solder bridge to enable disable internal motion control (like v1 scematics). It's one of the nice features of this chip, and it's a tragedy to make it impossible to use, even if 3d printers are currently too dumb to use the feature.

**Comment by PKav89** [comment]
Well, according to a Watterott schematic https://github.com/watterott/SilentStepStick/blob/master/hardware/SilentStepStick-TMC5160_v14.pdf there is no pulldown resistor, so it seems to be possible to switch TMC5160 into motion controller just by cutting SD_MODE trace and leaving SD_MODE pin floating. I'll test it later.

**Comment by PKav89** [comment]
I just tried cutting SD_MODE trace leaving SD_MODE floating and it seems to work.

IOIN before cut (tied to VCC_IO) 0x30:00:00:52
IOIN	after cut (SD_MODE floating) 0x30:00:00:12

So, bit 6 in the IOIN register changed from 1 to 0, which, according to datasheet, is "SD_MODE (1=External step and dir source)". I will try motion controller mode soon.

**Comment by PKav89** [comment]
I just tested motion controller, and yes, it runs OK with SD_MODE pin floating. Just cut the small trace on the board and TMC5160 switches into motion controller mode.

**Comment by madhunm** [comment]
> I just tested motion controller, and yes, it runs OK with SD_MODE pin floating. Just cut the small trace on the board and TMC5160 switches into motion controller mode.

Can you please post pictures of which trace you cut?

**Comment by iniciando-66** [comment]
Greetings
I'm an electronics technician working in factory maintenance here in Brazil and I would like to ask my friends about the Bigtreeteck TMC5160 v1.2 drive; I have been using the A4988 and TB6600 to control 1 Nema stepper motor controlled by a PWM signal generator board, but I would like to improve all of this by migrating to this new platform; I set up a prototype using the arduino uno to generate the spi signals but there is no reaction from the engine, do I need to install some other program before, or is it ready for use? Observing with the microscope the pins 20 (Vcc_IO) 21 (SD_MODE) and 22 (SPI_MODE) are all connected together in the IC, in this version there is no possibility to configure the operation in standalone (pin SPI_MODE low and SD_MODE high)?
Thank you very much for your attention.

**Comment by PKav89** [comment]
I wrote about it one post back. Just cut the SD_MODE trace and driver switches to Motion Control mode.

**Comment by madhunm** [comment]
> > I just tested motion controller, and yes, it runs OK with SD_MODE pin floating. Just cut the small trace on the board and TMC5160 switches into motion controller mode.
> 
> Can you please post pictures of which trace you cut?

Please share pictures

**Comment by PKav89** [comment]
I don't have a good camera to take a photo of such small things.
https://imgur.com/dCaj8Oy
This tiny trace, the middle one.

**Comment by creatormir** [comment]
The place to cut. It helped me.
![The place to cut.1](https://user-images.githubusercontent.com/37747152/117833065-3a721a80-b2a0-11eb-94a0-8248408605a8.jpg)

![The place to cut.](https://user-images.githubusercontent.com/37747152/117831777-1c57ea80-b29f-11eb-9e47-8f2545b80c35.jpg)

**Comment by xukangmin** [comment]
On V1.3, PIN 21 SD_MODE on the TMC5160-WA is connected to the solder jumper on the top of the board. The resistor is connected to VCCIO by default. You can jump the other way, then it's connected to GND. See image below. 

Hope it helps anyone searching for solutions.

![image](https://user-images.githubusercontent.com/5414312/210465442-d8009093-cdb6-496d-a405-57393d06564e.png)

**Comment by psychuk** [comment]
@xukangmin just bought some of these and want them to work with SPI.
Could you tell us what jumpers need to be connected within your green circle?

**Comment by xukangmin** [comment]
You can either remove the resistor or solder the resistor to the other way.
Both should work.
